tests/feed/test_feed.py

```diff
@@ -0,0 +1,176 @@
+                                    status="Confirmed")
+    student_course.save()
+    return student_course
+
+
+@pytest.fixture
+def persist_second_student_course(persist_second_course, persist_second_user):
+    student_course = StudentCourses(student_id=persist_second_user,
+                                    teacher_course_id=persist_second_course,
+                                    status="Confirmed")
+    student_course.save()
+    return student_course
+
+
+@pytest.fixture
+def persist_post_for_first_course(persist_course, persist_user, persist_first_student_course):
+    post = Post(course_id=persist_course,
+                user_id=persist_user,
+                parent_post_id=None,
+                content="this is a root post by a student for first course")
+    post.save()
+    return post
+
+
+@pytest.fixture
+def persist_reply_for_first_course(persist_course, persist_teacher,
+                                   persist_post_for_first_course, persist_second_student_course):
+    post = Post(course_id=persist_course,
+                user_id=persist_teacher,
+                parent_post_id=persist_post_for_first_course,
+                content="this is a reply post by a teacher for message in first course")
+    post.save()
+    return post
+
+
+@pytest.fixture
+def persist_post_for_second_course_student(persist_second_course, persist_second_user):
+    post = Post(course_id=persist_second_course,
+                user_id=persist_second_user,
+                parent_post_id=None,
+                content="this is a root post by a second student for second course")
+    post.save()
+    return post
+
+
+@pytest.mark.django_db
+class TestPost:
+    def test_is_root_post(self, persist_post_for_first_course, persist_reply_for_first_course):
+        assert persist_post_for_first_course.is_root_post()
+        assert not persist_reply_for_first_course.is_root_post()
+
+    def test_post_is_only_reply_to_root_post(self, persist_user, persist_course,
+                                             persist_reply_for_first_course):
+        reply_to_reply = Post(course_id=persist_course, user_id=persist_user,
+                              parent_post_id=persist_reply_for_first_course, content="should be invalid")
+        with pytest.raises(ValidationError):
+            reply_to_reply.save()
+
+    def test_is_user_able_to_post_in_course(self, persist_user, persist_course):
+        assert not Post.is_user_able_to_post_in_course(persist_user, persist_course)
+        user_joined_group = StudentCourses(student_id=persist_user, teacher_course_id=persist_course,
+                                           status="Confirmed")
+        user_joined_group.save()
+        assert Post.is_user_able_to_post_in_course(persist_user, persist_course)
+
+    def create_post_for_user_not_related_to_course(self, persist_user, persist_course):
+        assert not Post.is_user_able_to_post_in_course(persist_user, persist_course)
```

Comment:
what is invalid here? the post itself, or the fact the user is not related to the course? the test isn't very clear

Reply:
User should not be able to post in course because the user is not signed to course.
In this test, it is first asserted first that the user is not part of course, and then a post for said user for this course is attempted to be saved.